src/CMakeLists.txt

```diff
@@ -51,6 +51,20 @@ target_link_libraries(
 add_library(mp-units::mp-units ALIAS mp-units)
 install(TARGETS mp-units EXPORT mp-unitsTargets)
 
+# C++ module
+if(CMAKE_EXPERIMENTAL_CXX_MODULE_CMAKE_API)
+    add_library(mp-units-modules STATIC)
+    add_library(mp-units::modules ALIAS mp-units-modules)
+    target_sources(
+        mp-units-modules PUBLIC FILE_SET CXX_MODULES BASE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}" FILES "mp-units.cpp"
+    )
+    target_link_libraries(mp-units-modules PRIVATE mp-units::mp-units)
+    target_include_directories(
+        mp-units-modules ${unitsAsSystem} INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/core/include>
```

Comment:
Why only `core` includes are added and not the rest of them? When we link with `mp-units` target that contains all the library shouldn't we also expose all the headers as well?

Reply:
This one is for `mp_units/quantity_spec_macro.h`.
The others, `mp_units/bits/external/hacks.h` and `mp_units/bits/fmt_hacks.h`, are not part of the API.

Reply:
> When we link with `mp-units` target that contains all the library shouldn't we also expose all the headers as well?

A consumer of the `mp_units` module shouldn't care about other headers besides `mp_units/quantity_spec_macro.h`.

Reply:
Maybe this file should be moved to a special subdirectory like `module_include` and only this path should be exposed in a module?

Reply:
What about the installation part? Did you disable copying of the remaining header files for the module build?

Reply:
I agree with moving it standalone.
That way, a consumer of the module `mp_units` wouldn't be able to include other `core` headers.
But `core` would need this header, as I moved it out of `quantity_spec.h`.
However, then we'd also have to update compiler-explorer.

Reply:
> What about the installation part? Did you disable copying of the remaining header files for the module build?

To do that, and still support consumers of the header project, there'd be some complications.
I think it's just better to continue providing the header library regardless of support for C++ modules.

Reply:
I will update Compiler Explorer after merge. No worries.

Reply:
> To do that, and still support consumers of the header project, there'd be some complications.

What is the issue here? Can't we do it with branching in CMake?

Reply:
I won't know until I try it.
But I imagine we'd have to make something like
`find_package(mp_units COMPONENTS modules)` not provide the header-only library.
That may be doable.

Reply:
I am not so sure about it. I would like to get to a point that if I do:
- `cmake install -DHEADERS ...` I will see all the headers copied to the target directory
- `cmake install -DMODULE ...` I will have the BMI and only the `quantity_spec_macro.h` copied to the target directory

Of course, the name of the above defines is just for this example.

The same should be the case for Conan packages for both options (because there should be an option like `module: true/false` exposed from Conan).

Reply:
We'll see how that works.
Installing modules is not yet supported.
And you need to unreleased CMake v3.28.0 to consume imported modules.

I do remember that before, the plan was to install the sources, and have the consumers build the BMI.
But I think the recent talks suggest that the direction will be to install a BMI, as you suggest.